src/swap/core/getQuote.ts

```diff
@@ -0,0 +1,61 @@
+import { GetSwapQuote } from '../../definitions/swap';
+import { sendRequest } from '../../queries/request';
+import { formatToken } from '../../token/core/formatToken';
+import {
+  GetQuoteResponse,
+  GetQuoteParams,
+  GetQuoteParamsWithAddress,
+  Quote,
+  SwapError,
+  LegacyQuote,
+} from '../types';
+
+/**
+ * Retrieves a quote for a swap from Token A to Token B.
+ */
+export async function getQuote(params: GetQuoteParams): Promise<GetQuoteResponse> {
+  // Default parameters
+  const defaultParams = {
+    amountReference: 'from',
+  };
+
+  params = { ...defaultParams, ...params };
+
+  // Only pass in the address as a parameter to the request
+  // Defaults to ETH if no address is provided
+  const addressParams: GetQuoteParamsWithAddress = {
+    from: params.from.address || 'ETH',
+    to: params.to.address || 'ETH',
```

Comment:
Why the type can be wither an `Address` or a `string`? I guess the API works like this?